sktime/detection/base/_base.py

```diff
@@ -638,7 +638,8 @@ def predict_points(self, X):
         if task in ["anomaly_detection", "change_point_detection"]:
             return self._predict_points(X)
         elif task == "segmentation":
-            return self.segments_to_change_points(self.predict_segments(X))
+            segments = pd.DataFrame(self.predict_segments(X))
+            return self.segments_to_change_points(segments)
```

Comment:
> "In particular, segmentation tasks should have_predict return a `pd.Series` with an index of intervals, which is also specified in the `_predict_segments` method that calls `_predict`"

This is because `segments_to_change_points` expects a `pd.DataFrame` with an index of intervals, as specified in the docstrings: https://github.com/alyssadsouza/sktime/blob/9f52d772a9b9b10346ef0164ba8e413f47011304/sktime/detection/base/_base.py#L939

Reply:
You are right, the docstring type suggests `pd.DataFrame` for `segments_to_change_points`. Unfortunately, the text itself says `A series` and the example uses a `pd.Series`. I believe this was an oversight on my part. I would prefer keep the input for `segments_to_change_points` as `pd.Series` and update the incorrect docstring.

Reply:
@fkiraly, what are your thoughts.

Reply:
I agree with @tveten here: https://github.com/sktime/sktime/issues/7323#issuecomment-2507350221

that is, first `pd.Series` and mid-term `pd.DataFrame`.

I will try to move the outputs in bulk, and avoid breaking downstream, after merging this PR.